plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -37,27 +46,62 @@ public ProbabilisticSampler(TelemetrySettings telemetrySettings) {
         this.telemetrySettings = Objects.requireNonNull(telemetrySettings);
         this.samplingRatio = telemetrySettings.getSamplingProbability();
         this.defaultSampler = Sampler.traceIdRatioBased(samplingRatio);
+        this.actionBasedSampler = new HashMap<>();
+        this.actionBasedSamplingProbability = new HashMap<>();
     }
 
-    Sampler getSampler() {
+    /**
+     * Returns custom sampler based on the type of request
+     * @param attributes Telemetry attributes
+     */
+    Sampler getSampler(Attributes attributes) {
         double newSamplingRatio = telemetrySettings.getSamplingProbability();
         if (isSamplingRatioChanged(newSamplingRatio)) {
             synchronized (this) {
                 this.samplingRatio = newSamplingRatio;
                 defaultSampler = Sampler.traceIdRatioBased(samplingRatio);
             }
         }
+
+        final String action = attributes.get(AttributeKey.stringKey(TRANSPORT_ACTION));
+        if (action != null && telemetrySettings.isActionSamplingOverrideSet(action)) {
```

Comment:
We should be able to define the something like that also action = internal:* in the settings so that all the internal actions can be sampled at one defined rate. In case specific internal action is also defined then it should take the precedence over this limit.

Reply:
Will explore a bit on this & will it take as an follow-up.